Parsers/ASimDns/Product parsers/vimDnsCiscoUmbrella.yaml

```diff
@@ -0,0 +1,113 @@
+Parser:
+  Title: Cisco Umbrella DNS Events
+  Version: '0.0'
+  LastUpdated: May 10, 2021
+Product:
+  Name: Cisco
+Normalization:
+  Schema: Dns
+  Version: '0.2.0'
+References:
+- Title: ASIM DNS Schema
+  Link: https://aka.ms/AzSentinelDnsDoc
+- Title: ASIM
+  Link: https://aka.ms/AzSentinelNormalization
+Description: |
+  This is a Query Parser that is used to map Cisco Umrella Dns Events (Cisco_Umbrella_dns_CL) to the Azure Sentinel Information Model Dns schema.
+ParserName: vimDnsCiscoUmbrella
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr
+    Type: string
+    Default: '*'
+  - Name: domain_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: responsecodename
+    Type: string
+    Default: '*'
+  - Name: dnsresponsename
+    Type: string
+    Default: '*'
+  - Name: response_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventtype
+    Type: string
+    Default: 'lookup'
+ParserQuery: |
+  let DNSQuery_CiscoUmbrella=(
+    starttime:datetime=datetime(null), endtime:datetime=datetime(null)
+    , srcipaddr:string='*'
+    , domain_has_any:dynamic=dynamic([]) 
+    , responsecodename:string='*', dnsresponsename:string='*'
+    , response_has_any:dynamic=dynamic([]) , eventtype:string='lookup'
+    ){
+      Cisco_Umbrella_dns_CL
+      // ******************************************************************
+      //  Pre-parsing filterring:
+        | where
+          (isnull(starttime) or TimeGenerated > starttime)
+          and (isnull(endtime) or TimeGenerated < endtime) 
+          and (srcipaddr=='*' or InternalIp_s==srcipaddr)
+          and (array_length(domain_has_any) ==0 or Domain_s has_any (domain_has_any))
+          and (responsecodename=='*' or ResponseCode_s=~responsecodename)
+       // and (dnsresponsename=='*' or has_ipv4(ResponseName,dnsresponsename))
```

Comment:
Why comments? not available in Umbrella? It raises the question of how to document this.

Reply:
yes, not available, AFAIU
https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/CiscoUmbrella/Cisco_Umbrella
it is a documentation challenge, and also: do we return "all" or "none" as a match? (currently - all)